docs/_docs/04_reference/config-blocks-and-attributes.md

```diff
@@ -225,6 +225,34 @@ The `remote_state` block supports the following arguments:
     }
     ```
 
+Note that `remote_state` can also be set as an attribute. This is useful if you want to set `remote_state` dynamically.
+For example, if in `common.hcl` you had:
+
+```hcl
+locals {
+  remote_state = {
+    backend = "s3"
+    config = {
+      bucket = "mybucket"
+      key    = "path/to/my/key"
+      region = "us-east-1"
+    }
+  }
+}
+```
```

Comment:
Side note: It isn't strictly necessary to set this as a local, as `read_terragrunt_config` will expose the `remote_state` block under the `remote_state` attribute.

Reply:
Oh, interesting!

What about `generate` blocks? Is there a way to access those from what `read_terragrunt_config` returns?

Reply:
Generate is also available under `generate` attribute as a map. See https://github.com/gruntwork-io/terragrunt/blob/master/test/integration_test.go#L3043

Reply:
Oh, very handy. OK, docs updated: https://github.com/gruntwork-io/terragrunt/pull/1639/commits/675fc22446876104c2ccab49523c1c6e4020d10c.